Add ogr2ogr and ogrinfo (#3337)
tldr-pages/tldr: pages/common/ogr2ogr.md

# ogr2ogr

> Convert Simple Features data between file formats.
> More information: <https://gdal.org/programs/ogr2ogr.html#ogr2ogr>.

- Convert a Shapefile into a GeoPackage:

`ogr2ogr -f GPKG {{output}}.gpkg {{input}}.shp`

> - Reproject a GeoPackage from `EPSG:4326` to `EPSG:3857` coordinate reference system:

Reviewer: ```suggestion
- Reproject a GeoPackage changing coordinate reference system from `EPSG:4326` to `EPSG:3857`:
```

Author: I changed the wording to:

`- Change coordinate reference system of a GeoPackage from `EPSG:4326` to `EPSG:3857``

Author: @mebeim do you agree on my changed wording?

Reviewer: Oh, sorry I missed this comment, yes it looks fine to me, good job :)

> - Change coordinate reference system of a GeoPackage from `EPSG:4326` to `EPSG:3857`:

Author: I adapted the wording here

`ogr2ogr -s_srs EPSG:4326 -t_srs EPSG:3857 -f GPKG output.gpkg input.gpkg`

> - Convert a CSV file into a GeoPackage and assign projection:

Reviewer: ```suggestion
- Convert a CSV file into a GeoPackage, specifying the names of the coordinate columns and assigning a coordinate reference system:
```
Better clarify and also explains what all parameters are.

> `ogr2ogr -f GPKG output.gpkg input.csv -oo X_POSSIBLE_NAMES=lon -oo Y_POSSIBLE_NAMES=lat -a_srs 'EPSG:4326'`

Reviewer: ```suggestion
`ogr2ogr -f GPKG {{output}}.gpkg {{input.csv}} -oo {{X_POSSIBLE_NAMES=longitude}} -oo {{Y_POSSIBLE_NAMES=latitude}} -a_srs {{'EPSG:4326'}}`
```

Add user token delimiters. Also rename `lon` to `longitude` and `lat` to `latitude` for clarity.

- Load a GeoPackage into a PostGIS database:

> `ogr2ogr -f "PostgreSQL" PG:dbname="my_database" input.gpkg`

Reviewer: ```suggestion
`ogr2ogr -f "PostgreSQL" PG:dbname="{{database_name}}" {{input}}.gpkg`
```
Add user token delimiters. Also change `my_database` into `database_name`.

> `ogr2ogr -f GPKG {{output}}.gpkg {{input}}.csv -oo X_POSSIBLE_NAMES={{lon}} -oo Y_POSSIBLE_NAMES={{lat}} -a_srs {{EPSG:4326}}`

Reviewer: ```suggestion
`ogr2ogr -f GPKG {{output}}.gpkg {{input}}.csv -oo X_POSSIBLE_NAMES={{longitude}} -oo Y_POSSIBLE_NAMES={{latitude}} -a_srs {{EPSG:4326}}`
```

> - Filter file by bounding box (<xmin> <ymin> <xmax> <ymax>):

Reviewer: ```suggestion
- Clip layers of a GeoPackage file to the given bounding box:
```
"Filter" is not very accurate, what this command does is clipping. We can also move the names of the values in the command itself to make it both easier to read and to understand.

> `ogr2ogr -spat -13.931 34.886 46.23 74.12 -f GPKG output.gpkg input.gpkg`

Reviewer: ```suggestion
`ogr2ogr -spat {{min_x}} {{min_y}} {{max_x}} {{max_y}} -f GPKG {{output}}.gpkg {{input}}.gpkg`
```
Add user token delimiters. Also directly use names instead of values.

> `ogr2ogr -f "PostgreSQL" PG:dbname="{{my_database}}" {{input}}.gpkg`

Reviewer: ```suggestion
`ogr2ogr -f "PostgreSQL" PG:dbname="{{database_name}}" {{input}}.gpkg`
```

> - Filter GeoPackage by bounding box (<xmin> <ymin> <xmax> <ymax>):

Reviewer: ```suggestion
- Clip layers of a GeoPackage file to the given bounding box:
```

> `ogr2ogr -spat {{-13.931}} {{34.886}} {{46.23}} {{74.12}} -f GPKG {{output}}.gpkg {{input}}.gpkg`

Reviewer: ```suggestion
`ogr2ogr -spat {{min_x}} {{min_y}} {{max_x}} {{max_y}} -f GPKG {{output}}.gpkg {{input}}.gpkg`
```

# ogr2ogr

> > Convert simple features data between file formats.

Reviewer 2: What does "features data" mean in this context? It doesn't make grammatical sense.

Reviewer: I think it's the kind of data that these file formats contain: https://gdal.orgprograms/ogr2ogr.html it sounds strangely generic, so basically just how such kind of data is called 🤔🤔🤔

Author: [Simple Features](https://en.wikipedia.org/wiki/Simple_Features) are a specification how to store geographic data. A "feature" can be a point, line or polygon. It is a well known phrase in the domain of geodata. Let me know if I shall clarify more.

Reviewer 2: Ah, I see. In that case, I think it would make sense to capitalise it.
```suggestion
> Convert Simple Features data between file formats.
```

Author: done in bedb542cf10d8748cf130e7d72ccb0fb4631e43a

> More information: <https://gdal.org/programs/ogr2ogr.html#ogr2ogr>.

- Convert a Shapefile into a GeoPackage:

`ogr2ogr -f GPKG {{output.gpkg}} {{input.shp}}`

- Reproject a GeoPackage transforming coordinates from `EPSG:4326` to `EPSG:3857` :

`ogr2ogr -s_srs {{EPSG:4326}} -t_srs {{EPSG:3857}} -f GPKG {{output.gpkg}} {{input.gpkg`}}

- Convert a CSV file into a GeoPackage, specifying the names of the coordinate columns and assigning a coordinate reference system:

`ogr2ogr -f GPKG {{output.gpkg}} {{input.csv}} -oo X_POSSIBLE_NAMES={{lon}} -oo Y_POSSIBLE_NAMES={{lat}} -a_srs {{EPSG:4326}}`

- Load a GeoPackage into a PostGIS database:

`ogr2ogr -f "PostgreSQL" PG:dbname="{{my_database}}" {{input.gpkg}}`

- Filter file by bounding box (<xmin> <ymin> <xmax> <ymax>):

> `ogr2ogr -spat -13.931 34.886 46.23 74.12 -f GPKG {{output.gpkg}} {{input.gpkg}}`

Reviewer 3: These coordinates should also be within tokens.

Author: done in bedb542cf10d8748cf130e7d72ccb0fb4631e43a
